0000-catalog.md

```diff
@@ -0,0 +1,744 @@
+are supported:
+
+**catalog actions:**
+* CatalogCreate - create a catalog and store it in state
+* CatalogUpdate - update a catalog in state
+* *CatalogDelete - deletes a catalog from state and associated catalog_products
+
+
+**catalog_product actions:**
+* CatalogProductCreate - create a catalog_product and stores it in state
+* CatalogProductUpdate - updates a catalog_product in state
+* CatalogProductDelete - delete a catalog_product from state
+
+**catalog_product operation:**
+* *SetCatalogProductStatus - changes the "status" of a catalog_product to an active, inactive, or discontinued state
+
+The catalog_product actions/operation will accept a parameter list of the catalogs the operation should be performed on. Meaning Activate, Deactivate, or Discontinue can be performed on a single, multiple, or all catalogs that an organization has. 
+
+_`*` indicates that the action/operation can affect more than one catalog_product._
+
+## Permissions
+
+Creation of a Grid Catalog is restricted to agents acting on behalf of the
+organization in the catalog's owner field.  An organization must have the GS1
+company prefix in the "gs1_company_prefixes" metadata field for its Pike
+organization.  (Organization-level metadata will need to be implemented in
+Pike.) When a catalog is created, its owning organization is stored with the
+catalog in an "owner" field.
+
+Deletion of a catalog is restricted to agents acting on behalf of the
+organization in the catalog's owner field.  A setting will turn off deletion
+entirely, with the intent that it could be enabled/disabled by a Grid
+administrator.  Disabling delete is useful because external systems may have
+references to these products and deleting them could leave dangling references.
+
+*Disclaimer: CatalogDelete & CatalogProductDelete are potentially hazardous operation and need to be done with care.*
+
+
+To perform any of the **catalog actions** agents will need the following 
+permissions:
+```
+CatalogCreate: Agent from org needs the "can_create_catalog" permission
+CatalogUpdate: Agent from org needs the "can_update_catalog" permission
+CatalogDelete: Agent from org needs the "can_delete_catalog" permission
+```
+
+To perform any of the **catalog_product actions** agents inherit the same 
+permissions needed for a Grid Product:
+```
+CatalogProductCreate: Agent from org needs the "can_create_product" permission
+CatalogProductUpdate: Agent from org needs the "can_update_product" permission
+CatalogProductDelete: Agent from org needs the "can_delete_product" permission
+```
+
+In the case of **catalog operations** agents from the org they represent will
+need the operational permission to perform the desired operation on a Grid
+Catalog.
+```
+ActivateCatalogProduct: Agent from org needs the "can_activate_product_in_catalog" permission
+DeactivateCatalogProduct: Agent from org needs the "can_deactivate_product_in_catalog" permission
```

Comment:
Agreed

Reply:
Consolidated to 1 permission

Reply:
This comment was not a change request

Reply:
Gotcha. This could be better consolidated as one permission. If a higher level of granularity around permissions for adjusting a product's status is required, we can update it in a future iteration. For now, I think it's fine